### PR TITLE
Parse the annotated JDK as a stub file and remove jdk8.jar

### LINKED ISSUE #2943: Always use annotated JDK 11, even when using --release 8 or JDK 8
As a follow up to #2878.  

The annotated JDK 11 will be read as stub files.

### PR DESCRIPTION
**Pull request by smillst**
Fixes #2943.
Merge with: https://github.com/typetools/checker-framework-inference/pull/158.
TODO:
- [x] Finish addressing review comments.
- [x] Remove `checker/jdk` before merging this pull request.  (I haven't done this so that it is easier to see the changes in this pull request.)
- [x] Add change log.
- [X] Redo https://github.com/typetools/checker-framework-inference/pull/157. this with the correct branch name.
- [x] ~~Move the new `parseAllJdk` option to a different pull request and add a test that used it.~~ Merge https://github.com/typetools/checker-framework/pull/3279
- [x] Fix `checker-framework.demos` (and possibly other downstream tests).
- [x] Remove more methods from SystemUtil that use jdk.jar and update CFI.
- [X] Merge https://github.com/typetools/checker-framework.demos/pull/17.

### CODE REVIEW

File: framework/src/main/java/org/checkerframework/framework/source/SourceChecker.java [hunk 1]
```diff
@@ -356,7 +356,10 @@
 
     // Whether to output resource statistics at JVM shutdown
     // org.checkerframework.framework.source.SourceChecker.shutdownHook()
-    "resourceStats"
+    "resourceStats",
+
+    // Parse all JDK files at startup rather than as needed.
+    "parseAllJdk"
```
Comment (wmdietl) [inline framework/src/main/java/org/checkerframework/framework/source/SourceChecker.java:362]: This new option should be documented in the manual.

Comment (smillst) [reply framework/src/main/java/org/checkerframework/framework/source/SourceChecker.java:362]: This was done in #3279.

File: docs/manual/introduction.tex [hunk 2]
```diff
@@ -553,12 +553,9 @@
...
 % distributed version of the library --- they behave identically.
 
-There are two special cases.
+There is one special case.
 \begin{itemize}
 \item
-The annotated JDK is automatically put on your
-classpath; you don't have to do anything special for it.
-\item
 For the Javadoc classes in the JDK's \<com.sun.javadoc> package,
```
[the remark this answers is not in the document]
Comment (smillst) [reply docs/manual/introduction.tex:560]: Edit to say `For Java 8` add comment to `javadoc.astub` about the classes only be for Java 8.

File: changelog.txt [hunk 3]
```diff
@@ -1,5 +1,13 @@
 ---------------------------------------------------------------------------
 
+Version 3.4.0, May 1, 2020
+
+The annotated jdk8.jar is no longer used.  You can remove any occurrence of
```
Comment (smillst) [inline changelog.txt:5]: `You should`

**Pull request merged**